From f05f9b00d68bf4723d21a8d04509028923f447ce Mon Sep 17 00:00:00 2001
Message-ID: <f05f9b00d68bf4723d21a8d04509028923f447ce.1746781039.git.lorenzo@kernel.org>
In-Reply-To: <b840c5d58386d4600544ec7e02c2c3e19ce96bc6.1746781039.git.lorenzo@kernel.org>
References: <b840c5d58386d4600544ec7e02c2c3e19ce96bc6.1746781039.git.lorenzo@kernel.org>
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Wed, 30 Oct 2024 17:48:32 +0800
Subject: [PATCH 018/148] mtk: mt76: mt7996: fix key used-after-free issue

Fix the issue that the key might be used in TX path after being freed.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 mt7996/mac.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mt7996/mac.c b/mt7996/mac.c
index 341bc35c..0c1fdd5b 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -764,8 +764,7 @@ mt7996_mac_write_txwi_80211(struct mt7996_dev *dev, __le32 *txwi,
 	    info->flags & IEEE80211_TX_CTL_USE_MINRATE)
 		val |= MT_TXD1_FIXED_RATE;
 
-	if (key && multicast && ieee80211_is_robust_mgmt_frame(skb) &&
-	    key->cipher == WLAN_CIPHER_SUITE_AES_CMAC) {
+	if (key && multicast && ieee80211_is_robust_mgmt_frame(skb)) {
 		val |= MT_TXD1_BIP;
 		txwi[3] &= ~cpu_to_le32(MT_TXD3_PROTECT_FRAME);
 	}
-- 
2.49.0

